elastic/kibana
# [ResponseOps][Alerting] Register anomaly detection and custom threshold rule types under stack alerts feature privilege

## Summary

In the ES query, anomaly detection, and custom threshold rule types users can use the "Role visibility" dropdown to select where the rules should be accessible. The "Role visibility" dropdown sets the `consumer` which is paramount for alerting RBAC. For the anomaly detection and custom threshold rule types if the `consumer` is set to `stackAlerts` then the rules will not be accessible from any rule page even if the user has access to the "Stack alerts" feature privilege. This PR fixes this bug.

Fixes https://github.com/elastic/kibana/issues/193549
Fixes https://github.com/elastic/kibana/issues/191075
Fixes https://github.com/elastic/kibana/issues/184422
Fixes https://github.com/elastic/kibana/issues/179082

## Testing

1. Create an anomaly detection and custom threshold rule and set the "Role visibility"  to "Stack alerts".
2. Create a user with access only to "Stack alerts".
3. Login with the user created in Step 2.
4. Verify that you can see the rules from the stack management page.
5. Verify that you can see the alerts generated from the rules.
6. Create a user with roles `kibana_admin` and verify the same.


### Checklist


### For maintainers


## Release notes

Fix bug where rule types with "Stack alerts" role visibility are not being shown in the stack management page
